src/main/scala/org/broadinstitute/dsde/firecloud/trial/ProjectManager.scala

```diff
@@ -0,0 +1,153 @@
+package org.broadinstitute.dsde.firecloud.trial
+
+import akka.actor.{Actor, Props}
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.firecloud.FireCloudConfig
+import org.broadinstitute.dsde.firecloud.dataaccess.{HttpGoogleServicesDAO, RawlsDAO, TrialDAO}
+import org.broadinstitute.dsde.firecloud.model.Trial.{CreateProjectsResponse, CreationStatuses}
+import org.broadinstitute.dsde.firecloud.model.{AccessToken, WithAccessToken}
+import org.broadinstitute.dsde.firecloud.trial.ProjectManager.{Create, StartCreation, Verify}
+import org.broadinstitute.dsde.rawls.model.RawlsBillingProjectName
+import spray.http.OAuth2BearerToken
+
+import scala.concurrent.duration._
+import scala.util.Try
+
+object ProjectManager {
+
+  val DefaultCreateDelay = 2.minutes // pause between creating projects to avoid triggering Google quotas
+  val DefaultVerifyDelay = 3.minutes // pause between creating a project and verifying it, to allow Google to finish creation
+
+  sealed trait ProjectManagerMessage
+  case class StartCreation(count: Int) extends ProjectManagerMessage
+  case class Create(current: Int, target: Int) extends ProjectManagerMessage
+  case class Verify(projectName: String) extends ProjectManagerMessage
+
+  def props(rawlsDAO: RawlsDAO, trialDAO: TrialDAO,
+            createDelay: FiniteDuration = DefaultCreateDelay, verifyDelay: FiniteDuration = DefaultVerifyDelay): Props =
+    Props(new ProjectManager(rawlsDAO, trialDAO, createDelay, verifyDelay))
+}
+
+/**
+  * Actor to manage creation of free trial billing projects
+  *
+  * @param rawlsDAO dao for rawls
+  * @param trialDAO dao for the trial pool
+  * @param createDelay pause between creating projects to avoid triggering Google quotas
+  * @param verifyDelay pause between creating a project and verifying it, to allow Google to finish creation
+  */
+class ProjectManager(val rawlsDAO: RawlsDAO, val trialDAO: TrialDAO, val createDelay: FiniteDuration, val verifyDelay: FiniteDuration)
+  extends Actor with LazyLogging {
+
+  import context.dispatcher
+
+  val billingAcct = FireCloudConfig.Trial.billingAccount
+
+  // these should probably be an enum or the like
+  val idle = "idle"
+  val creating = "creating"
+
+  var currentStatus = idle
+
+  override def receive: Receive = {
+    case StartCreation(count: Int) => sender ! startCreation(count)
+    case Create(current: Int, target: Int) => sender ! create(current, target)
+    case Verify(projectName: String) => verify(projectName)
+  }
+
+  private def startCreation(count: Int): CreateProjectsResponse = {
+    getCurrentStatus match {
+      case `idle` =>
+        self ! Create(1, count)
+        CreateProjectsResponse(success = true, count, Some(s"$count projects are queued for creation."))
+      case x =>
+        CreateProjectsResponse(success = false, 0, Some("ProjectManager is already creating projects; don't try to create more!"))
```

Comment:
this "already creating" protection only works for the current orch instance. Since orch is behind a load balancer, it is possible for an end user to start both orch instances on creation loops ... potentially causing quota problems. Solving that is future work, not attempting to deal with it in this PR.

Reply:
Agreed - this is my concern as well as multiple trial managers (if that's a thing) creating projects at the same time.